app/modules/node-app-service/main.tf

```diff
@@ -37,14 +37,14 @@ resource "azurerm_app_service" "app_service" {
     {
       APPINSIGHTS_INSTRUMENTATIONKEY             = var.app_insights_instrumentation_key
       APPLICATIONINSIGHTS_CONNECTION_STRING      = var.app_insights_connection_string
-      ApplicationInsightsAgent_EXTENSION_VERSION = "~2"
+      ApplicationInsightsAgent_EXTENSION_VERSION = "~3"
       DOCKER_REGISTRY_SERVER_PASSWORD            = var.container_registry_server_password
       DOCKER_REGISTRY_SERVER_URL                 = var.container_registry_login_server
       DOCKER_REGISTRY_SERVER_USERNAME            = var.container_registry_server_username
       XDT_MicrosoftApplicationInsights_Mode      = "default"
       XDT_MicrosoftApplicationInsights_NodeJS    = "1"
     },
-    var.app_type == "frontend" ? {
+    var.inbound_vnet_connectivity == false ? {
```

Comment:
I think I only wanted this to apply to frontends rather than anything with an inbound connection! However, I think I want to change this approach entirely anyway, so think this can just stay for now!

Reply:
We could output it in all cases, it's just that when a private endpoint is assigned (when `inbound_vnet_connectivity` is `true`) then the public _app_.azurewebsites.net redirects to _app_.privatelink.azurewebsites.net instead, thus making it unresolvable from the Internet. Internally for resources in the same VNET the privatelink.azurewebsites.net DNS zone is made available via Azure Private DNS and is thus resolvable for other resources like App Gateway or other VNET-Integrated App Services, so it's still a valid address in those cases.

So I guess it may need changing if the app is relying on that `HOST_URL` value, because when App Gateway is implemented there probably won't be any apps where `inbound_vnet_connectivity` is `false`.

In any case, I'll leave it for now and we can review if it causes problems later 🙂